kolibri/plugins/learn/serializers.py

```diff
@@ -106,7 +107,7 @@ def get_assignments(self, instance):
         lesson_assignments = HierarchyRelationsFilter(LessonAssignment.objects.all()) \
             .filter_by_hierarchy(
                 target_user=current_user,
-                ancestor_collection=instance
+                ancestor_collection=F('collection')
         )
         filtered_lessons = Lesson.objects.filter(
             lesson_assignments__in=lesson_assignments,
```

Comment:
Should add `collection=instance` to the filtering for the `Lesson.objects` here to prevent lessons appearing in the wrong classroom (as we do for Exam objects below on line 124/5).